zoobot/pytorch/training/finetune.py

```diff
@@ -402,80 +429,112 @@ class FinetuneableZoobotClassifier(FinetuneableZoobotAbstract):
         num_classes (int): num. of target classes (e.g. 2 for binary classification).
         label_smoothing (float, optional): See torch cross_entropy_loss docs. Defaults to 0.
         class_weights (arraylike, optional): See torch cross_entropy_loss docs. Defaults to None.
-        
+
     """
 
     def __init__(
-            self,
-            num_classes: int,
-            label_smoothing=0.,
-            class_weights=None,
-            **super_kwargs) -> None:
+        self, num_classes: int, label_smoothing=0.0, class_weights=None, **super_kwargs
+    ) -> None:
 
         super().__init__(**super_kwargs)
 
-        logging.info('Using classification head and cross-entropy loss')
+        logging.info("Using classification head and cross-entropy loss")
         self.head = LinearHead(
             input_dim=self.encoder_dim,
             output_dim=num_classes,
-            dropout_prob=self.dropout_prob
+            dropout_prob=self.dropout_prob,
         )
         self.label_smoothing = label_smoothing
-        self.loss = partial(cross_entropy_loss,
-                            weight=class_weights,
-                            label_smoothing=self.label_smoothing)
-        logging.info(f'num_classes: {num_classes}')
+        self.loss = partial(
+            cross_entropy_loss,
+            weight=class_weights,
+            label_smoothing=self.label_smoothing,
+        )
+        logging.info(f"num_classes: {num_classes}")
         if num_classes == 2:
-            logging.info('Using binary classification')
-            task = 'binary'
+            logging.info("Using binary classification")
+            task = "binary"
         else:
-            logging.info('Using multi-class classification')
-            task = 'multiclass'
+            logging.info("Using multi-class classification")
+            task = "multiclass"
         self.train_acc = tm.Accuracy(task=task, average="micro", num_classes=num_classes)
         self.val_acc = tm.Accuracy(task=task, average="micro", num_classes=num_classes)
         self.test_acc = tm.Accuracy(task=task, average="micro", num_classes=num_classes)
-        
+
     def step_to_dict(self, y, y_pred, loss):
-        y_class_preds = torch.argmax(y_pred, axis=1) # type: ignore
-        return {'loss': loss.mean(), 'predictions': y_pred, 'labels': y, 'class_predictions': y_class_preds}
+        y_class_preds = torch.argmax(y_pred, axis=1)  # type: ignore
+        return {
+            "loss": loss.mean(),
+            "predictions": y_pred,
+            "labels": y,
+            "class_predictions": y_class_preds,
+        }
+
+    # Sanity check embeddings with linear evaluation first
+    def on_train_start(self) -> None:
+        with torch.no_grad():
```

Comment:
This makes sense for you and I but I'm not sure it should run by default for Zoobot. I think it may confuse astronomers who will now get two separate performance measurements for linear eval. I'm also a bit worried about the potential memory use from encoding *everything*. Maybe we can just wrap this in a new arg to ZoobotAbstract, linear_sanity_check=False, and an if statement/func here?

Reply:
Moved to if statement